### DISCUSSION TITLE
User Account "Watch" Flag (Moderation)

### DISCUSSION THREAD

**Discussion by CodeMouse92** [forem/forem#15321] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

I think most moderators know that feeling: you can sense from how a user conducts themselves that they're at a high risk of being toxic, but they seem to always stay *just* on the right side of the line. Sooner or later, it's inevitable that such a user will cross a line, and a moderator just has to hope they'll catch the problem when it happens...but too often, it escapes notice.

**Describe the solution you'd like**

A "Watch" flag that can be applied to a user account by moderators would be very helpful!

It would be a step below a full-blown flag, but would allow marking high-risk accounts. In addition to being a point of reference, this could increase how "eagerly" the automatic moderation algorithms evaluate the watched user's comments, increasing the probability that toxic behavior will be detected and caught.

Watch flags can also be set to age; if no comments are marked as "thumbs down" or "vomit" for an extended period of time (months), it can drop off. However, downvotes can also renew or increase the strength of a Watch flag. The Watch flag can also be "additive"; the more moderators mark "Watch" on the account, the more "eager" the algorithms become, and the longer it takes for the flag to age.

It would also be necessary for moderators to take back their own Watch flags, and for administrators to cancel flags manually.

**Describe alternatives you've considered**

Personally, when I notice such a user, I try to keep an eye on their comments, but I can't always remember to do so. I can alert other moderators to the same, but again, everyone must remember to manually monitor the user.

Naturally, a full-blown flag won't work here; since the user in question is usually "just to the right" of the line, there is nothing presently warranting action (besides perhaps an informal warning). The flag would then be expired, and the user's behavior is again at risk of evading radar.

**Additional context**

I would *not* recommend having a "Watched" badge on the user's comments, even if it were visible to moderators only! We don't want to get into a pattern of reading with a jaded eye and misinterpreting otherwise innocuous content.

However, it **would** be necessary to clearly mark (for moderators only) on user profiles, so as to not defeat the purpose of the feature.

However this is implemented, we want to make sure this does *not* mutate into a "scarlet letter".

**Comment by MarkoShiva** [comment]
Ok my problem is not related to that as modding is related to communities. I personally can't post a single comment anywhere or add new posts to reading list or liking them. 
However my old likes and old reading list stays. 

I have described issue here #5459 but no one is responded yet. I sent an email to yo@dev.to but ther eis no answer. In the meanwhile people are still following me and I can login but I cannot do anything useful like following people adding to reading list. commenting on posts.
Please someone check that out. @jessleenyc

**Comment by CodeMouse92** [comment]
@MarkoShiva Sorry you're having that experience, but leaving an off-topic comment on another issue like this won't speed this along; at worst, it will actually discourage response. Your other issue was only posted three days ago, so they'll get to it when they can. Unfortunately, you have to give it time. There's a LOT on Ben and Jess's plates.

**Comment by jessleenyc** [comment]
Hey @MarkoShiva, I believe this is an individual account issue - our support team will follow up with you separately via email.

**Comment by MarkoShiva** [comment]
Thanks @jessleenyc should I write them again on yo@dev.to I saw that email on bens contact information so I sent a email to DEV Community <yo@dev.to>
I hope someone will answer as I have that same problem already over 2 months. 

It would be nice if you guys would setup some service for support issues as right now in the same repo with all development issues they don't get noticed. Either add an other board for tickets on github or use youtrack or jira for tickets. 
Also there would be nice to have automated email answer when you receive a ticket. Also having some form for filling out the bug reports. 
Simple bug report form on platform would be pretty easy to implement and it would mean a lot for reporting this type of bugs. 
Thank you.

**Comment by CodeMouse92** [comment]
I don't want this feature suggestion to get lost in backlog because of the above tangent.

**Comment by cmgorton** [comment]
> I don't want this feature suggestion to get lost in backlog because of the above tangent.

@CodeMouse92 adding it to a new Github project I have created to get more visibility on this. Thanks for the issue.

**Comment by CodeMouse92** [comment]
Ever thanks, @cmgorton!

**Comment by cmgorton** [comment]
@michael-tharrington do you have any thoughts on this feature request. I think it could make a good potential RFC but wondering your thoughts on this feature for moderators.

**Comment by michael-tharrington** [comment]
I think this is a great idea all around! 

It's important to note that we have an internal `warned` status that we can set on users. It doesn't limit the user's abilities at all, but allows us to more easily track behavior — comments and posts — by these users. The way it currently works, I believe it feeds any warned user's comments and posts into a Slack channel. Ideally, it would feed these comments and posts into a queue that would be visible from our `/admin/feedback_messages` page. That page already has a queue for `latest articles` and `possible spam/abuse users` as well `abuse reports`. I could see us doing the same for `warned users` and listing out any content generate by these folks. 

With this in mind, I think we would extend the ability to mark a user as `warned` to our mods. 

There's likely still more thought to go into this. Do we display to all mods in some way who warned users are? We don't do this for admin currently unless they're looking at a user's admin page. Also, if we were to display who is warned, that would go against our general ethos of not publicly displaying disciplinary actions — BUT, if it's limited to mod view, then we may be willing to make that change. I do think it'd be helpful to have mod help keeping eyes on warned user behavior...

So, do we also feed the `warned user` content queue into `/mod` for mods to review? I think yes, but would like to hear other teammate's thoughts on this as it would be a pretty big change.

**Comment by CodeMouse92** [comment]
@michael-tharrington I agree, but a critical feature in this proposal is one additional status *below* warned, for those users who have not yet quite crossed the line (and therefore earned a warning), but whose content flies very close to that line. Some trolls are annoyingly good at staying just barely within the letter of the CoC. The idea is, when that user does cross the CoC, discovery and response is faster and more definitive.

**Comment by michael-tharrington** [comment]
That's a fair point. I think this needs just a bit more input from the team.

The current `warned `status that's only available to admin isn't as useful as it could be, so I thought possibly we could expand it to fit what you're talking about and avoid creating too many different layers of statuses. That said, when I think about it more, `warned`, really could just denote when we've warned someone to follow the CoC via email, and perhaps there should be an in-between status that is called `watched`.

I still think that we should think about many of the same questions I asked above. How will `watched` folks content be seen by mods (maybe in the mod center)? Will there be a queue in `admin/feedback` messages for admin to see `watched` folks content as well? Do we give mods and admin a visible flag in the community on `watched` users?
